src/lib/SX1280Driver/SX1280.cpp

```diff
@@ -0,0 +1,344 @@
+        break;
+    case SX1280_MODE_STDBY_XOSC:
+        hal.WriteCommand(SX1280_RADIO_SET_STANDBY, SX1280_STDBY_XOSC);
+        break;
+
+    case SX1280_MODE_FS:
+        hal.WriteCommand(SX1280_RADIO_SET_FS, 0x00);
+        break;
+
+    case SX1280_MODE_RX:
+
+        buf3[0] = 0x00; // periodBase = 1ms, page 71 datasheet, set to FF for cont RX
+        buf3[1] = 0xFF;
+        buf3[2] = 0xFF;
+        hal.WriteCommand(SX1280_RADIO_SET_RX, buf3, sizeof(buf3));
+        break;
+
+    case SX1280_MODE_TX:
+        //uses timeout Time-out duration = periodBase * periodBaseCount
+        buf3[0] = 0x00; // periodBase = 1ms, page 71 datasheet
+        buf3[1] = 0xFF; // no timeout set for now
+        buf3[2] = 0xFF; // TODO dynamic timeout based on expected onairtime
+        hal.WriteCommand(SX1280_RADIO_SET_TX, buf3, sizeof(buf3));
+        break;
+
+    case SX1280_MODE_CAD:
+        break;
+
+    default:
+        break;
+    }
+
+    currOpmode = OPmode;
+}
+
+void SX1280Driver::ConfigModParams(SX1280_RadioLoRaBandwidths_t bw, SX1280_RadioLoRaSpreadingFactors_t sf, SX1280_RadioLoRaCodingRates_t cr)
+{
+    // Care must therefore be taken to ensure that modulation parameters are set using the command
+    // SetModulationParam() only after defining the packet type SetPacketType() to be used
+
+    WORD_ALIGNED_ATTR uint8_t rfparams[3] = {0}; //TODO make word alignmed
+
+    rfparams[0] = (uint8_t)sf;
+    rfparams[1] = (uint8_t)bw;
+    rfparams[2] = (uint8_t)cr;
+
+    hal.WriteCommand(SX1280_RADIO_SET_MODULATIONPARAMS, rfparams, sizeof(rfparams));
+}
+
+void SX1280Driver::SetFrequency(uint32_t Reqfreq)
+{
+    //Serial.println(Reqfreq);
+    WORD_ALIGNED_ATTR uint8_t buf[3] = {0}; //TODO make word alignmed
+
+    uint32_t freq = (uint32_t)((double)Reqfreq / (double)SX1280_FREQ_STEP);
+    buf[0] = (uint8_t)((freq >> 16) & 0xFF);
+    buf[1] = (uint8_t)((freq >> 8) & 0xFF);
+    buf[2] = (uint8_t)(freq & 0xFF);
+
+    hal.WriteCommand(SX1280_RADIO_SET_RFFREQUENCY, buf, sizeof(3));
+    currFreq = Reqfreq;
+}
+
+int32_t SX1280Driver::GetFrequencyError()
+{
+    WORD_ALIGNED_ATTR uint8_t efeRaw[3] = {0}; //TODO make word alignmed
+    uint32_t efe = 0;
+    double efeHz = 0.0;
+
+    efeRaw[0] = hal.ReadRegister(SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MSB);
+    efeRaw[1] = hal.ReadRegister(SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MSB + 1);
+    efeRaw[2] = hal.ReadRegister(SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MSB + 2);
+    efe = (efeRaw[0] << 16) | (efeRaw[1] << 8) | efeRaw[2];
+
+    efe &= SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MASK;
+
+    //efeHz = 1.55 * (double)complement2(efe, 20) / (1600.0 / (double)GetLoRaBandwidth() * 1000.0);
+    return efeHz;
+}
+
+void SX1280Driver::SetFIFOaddr(uint8_t txBaseAddr, uint8_t rxBaseAddr)
+{
+    uint8_t buf[2];
+
+    buf[0] = txBaseAddr;
+    buf[1] = rxBaseAddr;
+    hal.WriteCommand(SX1280_RADIO_SET_BUFFERBASEADDRESS, buf, 2);
```

Comment:
This is sloppy sending '2' as the size. Unlikely, but what if someone comes along in future and adds another byte here? Or copy / pastes this as an example etc. Should be sizeof(buf)